DashAI/back/tasks/base_task.py

```diff
@@ -0,0 +1,171 @@
+                    param_choice = model_params[json_param].pop(
+                        "choice"
+                    )  # TODO See how to get the user choice
+                    param_class = filter_by_parent(
+                        model_json.get(json_param)["oneOf"][0].get("parent")
+                    ).get(param_choice)
+                    param_sub_params = parse_params(
+                        param_class.SCHEMA.get("properties"), model_params[json_param]
+                    )
+                    execution_params[json_param] = param_class(**param_sub_params)
+                else:
+                    execution_params[json_param] = model_params[json_param]
+            return execution_params
+
+        # TODO
+        # Remove models from the method signature
+        # Generate a Grid to search the best model
+        # self.executions: list = []
+        # for model, model_params in params.items():
+        execution = self.compatible_models[model]
+        model_json = execution.SCHEMA.get("properties")
+        # TODO use JSON_SCHEMA to check user params
+        execution_params = parse_params(model_json, param)
+        return execution(**execution_params)
+
+    def run_experiments(self, input_data: dict):
+        """
+        This method train all the executions in self.executions with the data in
+        input_data.
+        The input_data dictionary must have train and test keys to perform the training.
+        The test results were temporaly save in self.experimentResults.
+        """
+        # print("EXECUTIONS:")
+        # print(self.executions)
+        x_train = np.array(input_data["train"]["x"])
+        y_train = np.array(input_data["train"]["y"])
+        x_test = np.array(input_data["test"]["x"])
+        y_test = np.array(input_data["test"]["y"])
+
+        self.categories = []
+        for cat in y_train:
+            if cat not in self.categories:
+                self.categories.append(cat)
+        for cat in y_test:
+            if cat not in self.categories:
+                self.categories.append(cat)
+
+        numeric_y_train = []
+        for sample in y_train:
+            numeric_y_train.append(self.categories.index(sample))
+        numeric_y_test = []
+        for sample in y_test:
+            numeric_y_test.append(self.categories.index(sample))
+
+        for execution in self.executions:
+            execution.fit(x_train, numeric_y_train)
+
+            trainResults = execution.score(x_train, numeric_y_train)
+            testResults = execution.score(x_test, numeric_y_test)
+            parameters = execution.get_params()
+            # executionBytes = execution.save()
+
+            self.experimentResults[execution.MODEL] = {
+                "train_results": trainResults,
+                "test_results": testResults,
+                "parameters": parameters,
+                # " executionBytes": executionBytes,
+            }
+
```

Comment:
I think we should remove these, it shouldnt be the task's job to run executions and parse parameters